src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -737,4 +779,173 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
+     * If no grading scale exists, the user can only toggle the overall score median.
+     * If a grading scale exists, the user can switch between the overall score median and the median of the scores of all passed exams.
+     * Per default, the latter is selected in this case.
+     * @param medianType an enum indicating if the user toggles the overall median or the passed median
+     */
+    toggleMedian(medianType: MedianType): void {
+        let showMedian;
+        switch (medianType) {
+            case MedianType.PASSED:
+                this.showPassedMedian = showMedian = !this.showPassedMedian;
+                break;
+            case MedianType.OVERALL:
+            case MedianType.SUBMITTED:
+                this.showOverallMedian = showMedian = !this.showOverallMedian;
+                break;
+        }
+        if (showMedian) {
+            // At max only one median should be highlighted at a time. Therefore we deactivate the other one if a highlighting is selected
+            if (medianType === MedianType.PASSED) {
+                // The user selects the passed median to be highlighted, therefore we deactivate the highlighting of the other one
+                this.showOverallMedian = false;
+            } else {
+                // The user selects the overall median to be highlighted, therefore we deactivate the highlighting of the other one
+                this.showPassedMedian = false;
+            }
+            this.determineAndHighlightChartMedian(medianType);
+        } else {
+            this.activeEntries = [];
+        }
+    }
+
+    /**
+     * Auxiliary method that determines the median to be highlighted in the chart
+     * It identifies the bar representing the corresponding median type and
+     * highlights it by making all other chart bars a bit more transparent
+     * @param medianType enum representing the type of median to be highlighted
+     * @private
+     */
+    private determineAndHighlightChartMedian(medianType: MedianType): void {
+        let chartMedian;
+        if (medianType === MedianType.PASSED) {
+            chartMedian = this.aggregatedExamResults.medianRelativePassed ? roundScoreSpecifiedByCourseSettings(this.aggregatedExamResults.medianRelativePassed, this.course) : 0;
+            this.showPassedMedian = true;
+        } else {
+            this.setOverallChartMedianDependingOfExamsIncluded(medianType);
+            chartMedian = this.overallChartMedian;
+            this.showOverallMedian = true;
+        }
+        const index = this.findGradeStepIndex(chartMedian);
+        const medianChartBar = this.ngxData[index];
+
+        // Highlighting a bar only makes sense if this bar is representing a value > 0
+        if (medianChartBar.value === 0) {
+            if (medianType === MedianType.PASSED) {
+                this.showPassedMedianCheckbox = false;
+            } else {
+                this.showOverallMedianCheckbox = false;
+            }
```

Comment:
The idea is to hide the corresponding checkbox if the bar representing the median is 0 in order to prevent confusion. In my opinion it would be confusing to give the user the opportunity to toggle a highlighting that is not recognizable.
Therefore, this branching seems to be necessary to me, as depending of the median type that whose value is determined, we need to disable different checkboxes if the bar is 0.
I added some explanation comments to the different flags as well as to this code section that hopefully make it a bit clearer.

Reply:
That definitely makes sense. As noted in the edit, I missed the `CheckBox` suffix for the variables initially.